### PR TITLE
LPS-127222 Move code from asset_list/source.jsp

### PR DESCRIPTION
**Pull request by veroglez**

### CODE REVIEW

**Comment by veroglez** [comment]
ci:test:sf

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+					selectedItem.className,
+					selectedItem.name,
+					selectedItem.value,
+					selectedItem.displayValue,
+					`${selectedItem.label}: ${selectedItem.displayValue}`
+				);
+			},
+			selectEventName: `${namespace}selectDDMStructureField`,
+			title:
+				'<liferay-ui:message arguments="structure-field" key="select-x" />',
```
Comment (ealonso) [inline modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js:397]: @veroglez Creo que esto hay que cambiar por código js, Liferay.Language.get

Comment (veroglez) [reply modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js]: cambiado! gracias!

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+		if (structureOptions) {
+			structureOptions.classList.remove('hide');
+		}
+
+		const subtypeFieldsWrappers = document.querySelectorAll(
+			`#${namespace}${className}subtypeFieldsWrapper, #${namespace}${className}subtypeFieldsFilterEnableWrapper`
+		);
+
+		subtypeFieldsWrappers.forEach((subtypeFieldsWrapper) => {
+			if (selectedSubtype != 'false' && selectedSubtype != 'true') {
```
Comment (victorg1991) [inline modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js:131]: This is a bit weird, what values can selectedSubtype contain?

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+
+		if (fromBox.attr('id') === id || toBox.attr('id') === id) {
+			toggleSubclasses();
+		}
+	});
+
+	const openModal = ({delegateTarget}) => {
+		let url = delegateTarget.dataset.href;
+
+		url = Liferay.Util.addParams(
```
Comment (victorg1991) [inline modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js:364]: I think we can use URL class which has a nicer syntax

```
var url = new URL("http://foo.bar/?x=1&y=2");

url.searchParams.append('x', 42);

url.searchParams.set('x', 42);
```

Totally up to you if you want to change that

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+			url
+		);
+		url = Liferay.Util.addParams(
+			`${namespace}ddmStructureFieldValue=${encodeURIComponent(
+				ddmStructureFieldValueInput.value
+			)}`,
+			url
+		);
+
+		Liferay.Util.openSelectionModal({
```
Comment (victorg1991) [inline modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js:383]: I think all of these Liferay.Util.something can now be imported from frontend-js-web module so there is no need to use the global utilities here. I think the only exception is Liferay.Util.sub which I believe it is not exported

**Review by victorg1991 — COMMENTED** [review]
Hey I've left a couple of comments :) 

The pr looks good 👏  I've tried it and checked the linked bugs and everything works as expected

The thing that worries me more is about doing the same with the source.jsp file in the asset-publisher-web module, otherwise if some bug appear in this view we are going to do separate fixes for both files

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js (hunk 2 above)
Comment (veroglez) [reply modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js:131]: Yes, I agree with you that it is a bit weird, but it is. The values come in the following form:
- When you do not select anything, it's an empty value
- When you select a single type of article, it's an id
- When you select a multiple element type, it's a "true" or "false", it depends on the option.

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js (hunk 3 above)
Comment (veroglez) [reply modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js]: Great, I agree with you to change that

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js (hunk 4 above)
Comment (veroglez) [reply modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js]: I have imported only openSelectionModal and toggleDisabled. The others seem to be not in frontend-js-web yet.

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js (hunk 2 above)
Comment (veroglez) [reply modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/Source.js]: I have changed the "true" and "false" values to "all-subtypes" and "select-more-than-one", respectively.

**Comment by victorg1991** [comment]
Nice! :D

**Review by victorg1991 — APPROVED** [review]

**Comment by victorg1991** [comment]
resent here: https://github.com/liferay-echo/liferay-portal/pull/3728 :) thanks!!

**Pull request closed without merging**